tests/integ/test_large_query_breakdown.py

```diff
@@ -604,6 +607,31 @@ def test_optimization_skipped_with_views_and_dynamic_tables(session, caplog):
         Utils.drop_table(session, source_table)
 
 
+def test_large_query_breakdown_with_nested_select(session):
+    if not session.sql_simplifier_enabled:
+        pytest.skip(
+            "the nested select optimization is only enabled with sql simplifier"
+        )
+
+    temp_table_name = Utils.random_table_name()
+    final_df = create_df_with_deep_nested_with_column_dependencies(
+        session, temp_table_name, 8
+    )
+
+    with SqlCounter(query_count=1, describe_count=0):
+        queries = final_df.queries
+    assert len(queries["queries"]) == 3
+    assert queries["queries"][0].startswith("CREATE  SCOPED TEMPORARY  TABLE")
```

Comment:
Why double spaces here?

Reply:
that's how the snowflake queries are generated. it is not always perfect. For example here
https://github.com/snowflakedb/snowpark-python/blob/8dd23318a9d9ce55aa116d4b443d405bb165516d/src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py#L827-L831

notice there is an additional space on line 828. Such spaces issue creep-in many places but it doesn't affect sql generation.